crates/trigger/src/cli/variables.rs

```diff
@@ -0,0 +1,44 @@
+use spin_core::async_trait;
+use spin_factors::RuntimeFactors;
+use spin_factors_executor::ExecutorHooks;
+use spin_variables::{VariableProviderConfiguration, VariableSourcer};
+
+/// Implements TriggerHooks, sorting required variables
+pub struct VariableSorterExecutorHooks {
+    table: toml::Table,
+}
+
+impl VariableSorterExecutorHooks {
+    pub fn new(table: toml::Table) -> Self {
+        Self { table }
+    }
+}
+
+#[async_trait]
+impl<F: RuntimeFactors, U> ExecutorHooks<F, U> for VariableSorterExecutorHooks {
+    async fn configure_app(
+        &self,
+        configured_app: &spin_factors::ConfiguredApp<F>,
+    ) -> anyhow::Result<()> {
+        for (key, variable) in configured_app.app().variables() {
+            self.variable_env_checker(key.clone(), variable.clone())?;
+        }
+        Ok(())
+    }
+}
+
+impl VariableSourcer for VariableSorterExecutorHooks {
+    fn variable_env_checker(&self, key: String, val: spin_app::Variable) -> anyhow::Result<()> {
```

Comment:
`variable_env_checker` is a noun.  I think the idea here is something like `check_variable_has_value` or `check_variable_resolves`.